tests/30rooms/60version_upgrade.pl

```diff
@@ -205,22 +205,32 @@ sub upgrade_room_synced {
              log_if_fail "PL content in old room", $pl_content;
          }
       )->then( sub {
-         upgrade_room_synced(
-            $creator, $room_id,
-            new_version => $TEST_NEW_VERSION,
+         # we have to wait for the second PL event to turn up.
+         my $c = 0;
+         matrix_do_and_wait_for_sync(
+            $creator,
+            do => sub {
+               upgrade_room(
+                  $creator, $room_id, new_version => $TEST_NEW_VERSION,
+               );
+            },
+            check => sub {
+               my ( $sync_body, $new_room_id ) = @_;
+               return 0 if not exists $sync_body->{rooms}{join}{$new_room_id};
+               my $tl = $sync_body->{rooms}{join}{$new_room_id}{timeline}{events};
+               foreach my $ev ( @$tl ) {
+                  next if $ev->{type} ne 'm.room.power_levels';
+                  $c++;
+                  log_if_fail "PL event $c in new room", $ev;
+                  return $ev if $c == 2;
+               }
+               return 0;
+            },
```

Comment:
I'm a bit confused why this is needed. Can't this be in `upgrade_room_synced`? Do we send a two PL in the new room? It feels a bit useless to have a `upgrade_room_sync` if its not finished upgrading when it returns.

Reply:
Yes, we send two PL events in the new room.

The problem here is (a) knowing when the upgrade has finished (or, more to the point, all the resultant events have been synced back) and (b) given that those events could span more than one sync, returning them from `upgrade_room_synced` to the tests so that they can be verified.